src/scripts.js

```diff
@@ -53,12 +57,41 @@ allRecipesContainer.addEventListener('click', displayRecipe);
 taggedRecipesContainer.addEventListener('click', displayRecipe);
 weeklyRecipesButton.addEventListener('click', displayWeeklyRecipes);
 searchIngredientGlide.addEventListener('click', selectTag);
+favoritesViewButton.addEventListener('click', displayFavoritesView);
+favoriteRecipeButton.addEventListener('click', favoriteRecipe);
+favoriteRecipesContainer.addEventListener('click', displayRecipe);
+
+function favoriteRecipe(event) {
+  const favoriteRecipe = recipeRepository.recipes.find(recipe => recipe.id === parseInt(favoriteRecipeButton.name));
+
+  if (user.favoriteRecipes.includes(favoriteRecipe)) {
+    user.removeFavorite(favoriteRecipe);
+  } else {
+    user.addFavorite(favoriteRecipe);
+  }
```

Comment:
Based on this check here, I don't think you need the `if` check in `addRecipes`. 

Also, is there something you could do here to change the text of the button to say **"Remove"** instead of **"Add"**? Maybe you build a remove button and show/hide either button based on this check? 

In short, how is the user supposed to know that clicked a different color button that still says "Add" removes a favorite? 😄